src/snyk/snykCode/views/suggestion/codeSuggestionWebviewProvider.ts

```diff
@@ -304,6 +305,64 @@ export class CodeSuggestionWebviewProvider
           }
           break;
         }
+        case 'generateVulnerabilityExplanation': {
+          this.logger.info('generating vulnerability explanation');
+          const { suggestion } = message.args;
+          const filePath = suggestion.filePath;
+          const fileContent = readFileSync(filePath, 'utf8');
+
+          const derivationLineNumbers: Set<number> = new Set<number>();
+          for (const markerLocation of suggestion.markers!) {
+            for (const markerPos of markerLocation.pos) {
+              const lines = markerPos.rows;
+              for (const line of lines) {
+                derivationLineNumbers.add(line + 1);
+              }
+            }
+            markerLocation.pos;
+          }
+          console.log('derivation lines: ', ...derivationLineNumbers);
+
+          const derivationLines: string[] = [];
+          const fileLines: string[] = fileContent.split('\n');
+          for (const derivationLineNumber of derivationLineNumbers) {
+            derivationLines.push(fileLines.at(derivationLineNumber - 1)!);
+          }
+          let derivation = derivationLines.join(',');
+          derivation = derivation.replace(/\t/g, '  ');
+          console.log('derivation: ', derivation);
```

Comment:
Dunno what derivation is, but probably should be done on the LS side. Should just pass the issueID and the fixID and let LS do all the work.